File: frontend/src/lib/api/video-overviews.ts
import apiClient from './client'

export interface VideoOverviewComposeRequest {
  slide_deck_artifact_id: string
  podcast_episode_id: string
  caption_language?: string
}

export interface VideoOverview {
  artifact_id: string
  episode_id: string
  duration_seconds: number
  media_url: string
  captions_url: string
}

export const videoOverviewsApi = {
  compose: async (payload: VideoOverviewComposeRequest) => (
    await apiClient.post<VideoOverview>('/video-overviews', payload)
  ).data,
}

File: api/schemas/video_overviews.py
"""Contracts for local, source-grounded Video Overview composition."""

from __future__ import annotations

from pydantic import BaseModel, Field


class VideoOverviewComposeRequest(BaseModel):
    slide_deck_artifact_id: str = Field(min_length=1)
    podcast_episode_id: str = Field(min_length=1)
    caption_language: str = Field(
        default="en", pattern=r"^[a-zA-Z]{2,3}(?:-[a-zA-Z0-9]{2,8})*$"
    )


class VideoOverviewResponse(BaseModel):
    artifact_id: str
    episode_id: str
    duration_seconds: float = Field(gt=0)
    media_url: str
    captions_url: str
